annotation/tests/test_tasks_services.py

```diff
@@ -585,3 +636,1697 @@ def test_filter_tasks_db_no_files_or_jobs(
+                    ]
+                }
+            },
+            {
+                1: [
+                    (
+                        {
+                            "name": "Task 1",
+                            "details": {"id": "a1", "info": "Details 1"},
+                        },
+                        "1",
+                    ),
+                    ({"name": "Task 2"}, "2"),
+                ]
+            },
+        ),
+        (
+            {
+                2: {
+                    "objects": [
+                        {
+                            "id": "3",
+                            "info": {"id": "b3", "value": "Nested Value"},
+                        },
+                    ]
+                }
+            },
+            {2: [({"info": {"id": "b3", "value": "Nested Value"}}, "3")]},
+        ),
+        (
+            {
+                3: {
+                    "objects": [
+                        {"id": "4", "name": "Task 4"},
+                        {"id": "5", "status": "Complete"},
+                    ]
+                },
+                4: {
+                    "objects": [
+                        {"id": "6", "description": "Task 6"},
+                    ]
+                },
+            },
+            {
+                3: [({"name": "Task 4"}, "4"), ({"status": "Complete"}, "5")],
+                4: [({"description": "Task 6"}, "6")],
+            },
+        ),
+    ),
+)
+def test_get_tasks_without_ids(
+    tasks: Dict[int, Any], expected: Dict[int, Any]
+):
+    assert get_tasks_without_ids(tasks) == expected
+
+
+@pytest.fixture
+def mock_dependencies():
+    with patch(
+        "annotation.tasks.services.get_file_path_and_bucket",
+        return_value=("s3/path", "bucket"),
+    ) as mock_get_file_path_and_bucket, patch(
+        "annotation.tasks.services.get_annotation_tasks", return_value={}
+    ) as mock_get_annotation_tasks, patch(
+        "annotation.tasks.services.construct_annotated_pages",
+        return_value=([], set()),
+    ) as mock_construct_annotated_pages, patch(
+        "annotation.tasks.services.construct_annotated_doc", return_value=None
+    ) as mock_construct_annotated_doc, patch(
+        "annotation.tasks.services.update_task_status", return_value=None
+    ) as mock_update_task_status, patch(
+        "annotation.tasks.services.Logger.exception", return_value=None
+    ) as mock_logger_exception:
+
+        yield {
+            "mock_get_file_path_and_bucket": mock_get_file_path_and_bucket,
+            "mock_get_annotation_tasks": mock_get_annotation_tasks,
+            "mock_construct_annotated_pages": mock_construct_annotated_pages,
+            "mock_construct_annotated_doc": mock_construct_annotated_doc,
+            "mock_update_task_status": mock_update_task_status,
+            "mock_logger_exception": mock_logger_exception,
+        }
+
+
+def test_create_validation_revisions(mock_dependencies: Dict[str, Mock]):
+    mock_db = MagicMock()
```

Comment:
We have made a fixture for this, which you already used in previous tests. You can use it here too